dataikuapi/dss/modelcomparison.py

```diff
@@ -0,0 +1,164 @@
+
+    def get_object_discussions(self):
+        """
+        Get a handle to manage discussions on the model comparison
+
+        :returns: the handle to manage discussions
+        :rtype: :class:`dataikuapi.discussion.DSSObjectDiscussions`
+        """
+        return DSSObjectDiscussions(self.client, self.project_key, "MODEL_COMPARISON", self.mec_id)
+
+    ########################################################
+    # Deletion
+    ########################################################
+
+    def delete(self):
+        """
+        Delete the model comparison
+
+        """
+        return self.client._perform_empty("DELETE", "/projects/%s/modelcomparisons/%s" % (self.project_key, self.mec_id))
+
+
+class DSSModelComparisonSettings(object):
+    """
+    A handle on the settings of a model comparison
+
+    A model comparison has:
+    - a display name ;
+    - a prediction type ;
+    - a list of full ids of items to compare
+
+    The prediction type can be:
+    - BINARY_CLASSIFICATION,
+    - REGRESSION,
+    - MULTICLASS
+
+    The full ids are:
+    - the model id of a Lab Model,
+    - the model id of a saved model version,
+    - the model evaluation id of a model evaluation.
+
+
+    Do not create this class directly, instead use :meth:`dataikuapi.dss.DSSModelComparison.get_settings`
+    """
+    def __init__(self, model_comparison, settings):
+        self.model_comparison = model_comparison
+        self.settings = settings
+
+    def get_raw(self):
+        """
+        Get raw settings of a model comparison
+
+        :return: the raw settings of comparison, as a dict. Modifications made to the returned object
+        are reflected when saving
+        :rtype: dict
+        """
+        return self.settings
+
+    def add_compared_item(self, full_id):
+        """
+        Add an item to the list of compared items
+
+        :param full_id: full id of the item (lab model, saved model version, model evaluation) to add
+        """
+        if "comparedModels" not in self.settings:
+            self.settings["comparedModels"] = []
+        self.settings["comparedModels"].append({
+            "refId": full_id
+        })
+
+    def remove_compared_item(self, full_id):
+        """
+        Remove an item from the list of compared items
+
+        :param full_id: full id of the item (lab model, saved model version, model evaluation) to remove
+        """
+        if not self.settings["comparedModels"]:
+            return
+        self.settings["comparedModels"] = filter(lambda x: x["refId"] != full_id, self.settings["comparedModels"])
+
+    def get_compared_items(self):
+        """
+        Get the full ids of items compared in this comparison
+
+        :return: the list of the full ids of compared items
+        :rtype: list[str]
+        """
+        if not self.settings["comparedModels"]:
+            return []
+        return map(lambda x: x["refId"], self.settings["comparedModels"])
```

Comment:
```suggestion
        return [x["refId"] for x in self.settings["comparedModels"]]
```
I suggest you don't use `map` if you want to have the same behavior in python 2 and python 3 and to return a list in each case.

Reply:
Thanks for this suggestion, did not know of this discrepancy!